### DISCUSSION TITLE
After sleep mode in win10 audio in lazer completely disappears and beatmaps won't start

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#14727] [Q&A] [answered]
Just closed my laptop cover and see what's happened

https://user-images.githubusercontent.com/86934170/132996734-ac2fec80-eeb3-41f8-9f95-e158029114a9.mp4

**Comment by Loreos7** [comment]
![image](https://user-images.githubusercontent.com/86934170/133000890-946e9cde-4338-44e2-a329-e033ae4c72b7.png)
![image](https://user-images.githubusercontent.com/86934170/133000940-6e22a7fa-f4de-450c-b8fd-5407dc93de9b.png)

**Comment by frenzibyte** [comment]
Interestingly I can reproduce this on macOS, by indeed putting it to sleep for a short period. And changing the audio device from "Default" to the actual default device itself seems to fix it.

While in this broken state, samples that have been constructed for the first time post-breakage seem to play correctly, which may indicate that it's somewhat of a failure in audio device migration leaving cached audio components to no longer play correctly.

I don't have the time to investigate this currently but will see, if it's not easily reproducible.

**Comment by MishterKirby** [comment]
If I may chip in, I noticed that it switches from my HDMI output to my dedicated speakers before it finishes going to sleep, but I'm not sure if that's my hardware or Windows doing it, or if it's osu! switching between them.

**Reply by Loreos7** [reply]
For addition: stable works correctly with sleep

**Comment by Loreos7** [comment]
[performance.log](https://github.com/ppy/osu/files/7370924/performance.log)
[updater.log](https://github.com/ppy/osu/files/7370925/updater.log)
[database.log](https://github.com/ppy/osu/files/7370926/database.log)
[performance-draw.log](https://github.com/ppy/osu/files/7370927/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/7370928/performance-update.log)
[performance-audio.log](https://github.com/ppy/osu/files/7370929/performance-audio.log)
[network.log](https://github.com/ppy/osu/files/7370930/network.log)
[runtime.log](https://github.com/ppy/osu/files/7370932/runtime.log)

**Comment by peppy** [comment]
I'm pretty sure this isn't just related to sleep, but any loss of audio device. That was something that used to work but doesn't any more.

**Reply by Loreos7** [reply]
It has a solution by switching audio device instead of "default". But in osu!stable sleep mode is compatible with the game.

**Comment by Loreos7** [comment, marked as answer]
~After more than a year this problem is fixed itself somehow. Now everything works fine.~
https://github.com/ppy/osu/issues/14729

**Reply by Theighlin** [reply]
You can see an issue was created from this discussion, and that has been closed by a framework pr, so i don't think "fixed itself somehow" is correct here (sorry for bump)

**Reply by Loreos7** [reply]
Oh wait I didn't notice. Thank you for pointing me
